consensus/src/runway/backup.rs

```diff
@@ -139,43 +160,51 @@ fn load_backup<H: Hasher, D: Data, S: Signature, R: Read>(
         for parent_id in parent_ids.elements() {
             let parent = UnitCoord::new(coord.round() - 1, parent_id);
             if !already_loaded_coords.contains(&parent) {
-                return Err(LoaderError::InconsistentData(coord));
+                return Err(BackupReadError::InconsistentData(coord));
             }
         }
 
         already_loaded_coords.insert(coord);
     }
 
-    Ok(loaded_units)
+    Ok(loaded_items)
 }
 
 fn on_shutdown(starting_round_tx: oneshot::Sender<Option<Round>>) {
     if starting_round_tx.send(None).is_err() {
-        warn!(target: "AlephBFT-unit-backup", "Could not send `None` starting round.");
+        warn!(target: LOG_TARGET, "Could not send `None` starting round.");
     }
 }
 
-/// Loads Unit data from `unit_loader` and awaits on response from unit collection.
-/// It sends all loaded units by `loaded_unit_tx`.
-/// If loaded Units are compatible with the unit collection result (meaning the highest unit is from at least
+/// Loads backup data from `backup_reader` and awaits on response from unit collection.
+/// It sends all loaded items by `loaded_items_tx`.
+/// If loaded units are compatible with the unit collection result (meaning the highest unit is from at least
 /// round from unit collection + 1) it sends `Some(starting_round)` by
-/// `starting_round_tx`. If Units are not compatible it sends `None` by `starting_round_tx`
-pub async fn run_loading_mechanism<'a, H: Hasher, D: Data, S: Signature, R: Read>(
-    unit_loader: UnitLoader<R, H, D, S>,
+/// `starting_round_tx`. If items are not compatible it sends `None` by `starting_round_tx`
+pub async fn run_loading_mechanism<'a, H: Hasher, D: Data, MK: MultiKeychain, R: Read>(
```

Comment:
I can think of introducing separate functions for unit / alert / multisig validation. In fact they can all happen independently. Would that resolve? That would fit the next pr better tho as in it I'm going to implement backup loading

Reply:
There is also the problem that all this requires at least partial synchronicity – e.g. some units only get accepted if an appropriate alert has been confirmed. Since this method wants to send the starting round, this means here we _have_ to include loading alerts (unless I'm missing something ofc).